### ISSUE TITLE
consistency in logging images between multiple logging backends -- wandb and tensorboard

### ISSUE THREAD

**Issue by 00krishna** [PyTorchLightning/pytorch-lightning#8039]
## 🐛 Bug

I am still new to PL so perhaps there is a better way to do this. But there seems to be an issue with consistency when logging
model performance to different backends. I have a simple sequence-to-sequence model and I wanted to log some initial checks on the training data, to make sure that the data and labels line up correctly. I was logging to both tensorboard and weights and biases (wandb), through a custom callback. However I found that I needed to manually change the image logging call, depending on which backend I was using. That creates really fragile code, since if I change the logging backend callback--or even the order of callback for logging, then the code will break. 

## Please reproduce using the BoringModel

I have my model and a simple trainer class:
```
trainer = Trainer(logger=[tb_logger, wandb_logger])
```

Then I created a custom callback to the initial batch of images to both loggers. But as you can see I have to refer to `experiment[0]` and `experiment[1]` to log correctly. I think the correct way would be to use just `experiment.log()` and this should be agnostic of which backend is used. 

```
class TimeSeriesDataSanityCheckCallback(Callback):

    def on_train_batch_start(self, trainer, pl_module , batch, batch_idx, dataloader_idx) -> None:
        source, target = batch
        plots = []
        for i in range(len(source)):
            fig, (ax1, ax2) = plt.subplots(1, 2)
            fig.suptitle(f'Sensor data time-series {i}')
            ax1.plot(range(len(source[i, :, 0])), source[i, :, 0])
            ax2.plot(range(len(target[i, :, 0])), target[i, :, 0])
            im = plot_to_image(fig)
            plots.append(im)
        
        grid = torchvision.utils.make_grid(plots, nrow=1) 
        pl_module.logger.experiment[0].add_image('generated', grid, 0)
        pl_module.logger.experiment[1].log({"val_input_image":[wandb.Image(grid.cpu(), caption="val_input_image")]})
```
I have that auxiliary function `plot_to_image` to simple take matplotlib plots and convert them to tensors. Here is that code if it helps. 

```
class TimeSeriesDataSanityCheckCallback(Callback):

    def on_train_batch_start(self, trainer, pl_module , batch, batch_idx, dataloader_idx) -> None:
        source, target = batch
        plots = []
        for i in range(len(source)):
            fig, (ax1, ax2) = plt.subplots(1, 2)
            fig.suptitle(f'Sensor data time-series {i}')
            ax1.plot(range(len(source[i, :, 0])), source[i, :, 0])
            ax2.plot(range(len(target[i, :, 0])), target[i, :, 0])
            im = plot_to_image(fig)
            plots.append(im)
        
        grid = torchvision.utils.make_grid(plots, nrow=1) 
        pl_module.logger.experiment[0].add_image('generated', grid, 0)
        pl_module.logger.experiment[1].log({"val_input_image":[wandb.Image(grid.cpu(), caption="val_input_image")]})
```

### Expected behavior

The expected behavior is to have a consistent logging interface and not need to specialize the call depending on which loggers are used. That would be a more robust method. 

### Environment

```
wget https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py
# For security purposes, please check the contents of collect_env_details.py before running it.
python collect_env_details.py
```

 - PyTorch Version (e.g., 1.0): 1.9
 - OS (e.g., Linux): ubuntu linux 20.04.2
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.9
 - CUDA/cuDNN version: cudatoolkit 11.1
 - GPU models and configuration: 1 gpu
 - Any other relevant information:

**Comment by 00krishna** [comment]
Seems related to https://github.com/PyTorchLightning/pytorch-lightning/issues/3635

**Comment by ethanwharris** [comment]
Hi @00krishna thanks for reporting this issue. It may also he related to #6720 and #7920 - please feel free to shar your thoughts on those as well.

Thanks :smiley:

**Comment by tchaton** [comment]
Dear @00krishna,

Definitely a feature on our roadmap ! Would you be interested in making a PR following chat in  #6720 ?

Best,
T.C
